android/KMEA/app/src/test/java/com/keyman/engine/view/FunctionalTestHelper.java

```diff
@@ -66,7 +66,21 @@ static void installCustomKeyboard(File aKPMFile) throws IOException, JSONExcepti
 
     Assert.assertEquals(installedKbds.size(),1);
 
-    KMManager.addKeyboard(ApplicationProvider.getApplicationContext(),(HashMap<String, String>) installedKbds.get(0));
+    Map<String, String>firstKbd = installedKbds.get(0);
+    Keyboard installedKbd = new Keyboard(
+      firstKbd.get(KMManager.KMKey_PackageID),
+      firstKbd.get(KMManager.KMKey_KeyboardID),
+      firstKbd.get(KMManager.KMKey_KeyboardName),
+      firstKbd.get(KMManager.KMKey_LanguageID),
+      firstKbd.get(KMManager.KMKey_LanguageName),
+      firstKbd.get(KMManager.KMKey_KeyboardVersion),
+      firstKbd.containsKey(KMManager.KMKey_CustomHelpLink) ? firstKbd.get(KMManager.KMKey_CustomHelpLink) : "",
+      "",
+      true,
+      firstKbd.containsKey(KMManager.KMKey_Font) ? firstKbd.get(KMManager.KMKey_Font) : "",
+      firstKbd.containsKey(KMManager.KMKey_OskFont) ? firstKbd.get(KMManager.KMKey_OskFont) : "");
+
+    KMManager.addKeyboard(ApplicationProvider.getApplicationContext(), installedKbd);
```

Comment:
We need to fix the return value from `kmpProcessor.processKMP()` and also any other locations where a `Map<String,String>` is used, e.g. the events like `onPackageInstalled()` to use `Keyboard[]` rather than a `Map<String, String>`.

So... all that to say: can we split the deprecation part of this PR into a separate PR? Because it looks like we need to do more work on the API (and internals too), otherwise API consumers are going to have to do this kind of fragile refactoring.

Reply:
>  can we split the deprecation part of this PR into a separate PR?

I've split the old SDK calls to a separate #8612. When I resume this PR it'll be for addressing the API

Reply:
Sadness - the Map<String, String> ripples out to the other places like KeyboardEventHandler and CloudCatalog.. Too much refactoring to tackle now.